public/javascripts/controllers/buyer-ctrl.js

```diff
@@ -227,6 +230,89 @@ angular.module('jasmic.controllers')
                 $scope.demand.de_until = moment().add(7, 'days').toDate();
             };
 
+            $scope.editDemandForm = function (obj) {
+
+                $scope.edit_demand = !$scope.edit_demand;
+
+                $scope.demand = obj;
+
+                //for some reason the dates needed to be passed through MomentJS as seen below to be assigned to the form inputs
+
+                $scope.demand.de_posting_date= moment(obj.de_posting_date).toDate();
```

Comment:
@nick22891 and @tremainebuchanan `moment` attempt to convert the ISO-8601 date format that `mongodb` uses to javascript dates that browsers are used to handling.